node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/task/util/file/FileSync.java

```diff
@@ -64,9 +76,20 @@ private boolean maybeUpdateContent(TaskContext taskContext,
             return false;
         } else {
             taskContext.recordSystemModification(logger, "Patching file " + path);
-            path.writeBytes(content.get());
+            writeBytes(content.get(), atomicWrite);
             contentCache.updateWith(content.get(), currentAttributes.forceGet().lastModifiedTime());
             return true;
         }
     }
+
+    private void writeBytes(byte[] content, boolean atomic) {
+        if (atomic) {
+            String tmpPath = path.toPath().toString() + ".FileSyncTmp";
```

Comment:
Any reason to not use `Files.createTempFile`?

Reply:
I specifically did not want to use it, in case there is an exception (or restart of host-admin) after createTempFile and before the move: It would leave stray files in the directory.

Reply:
More importantly, the temp file may be written on a different file system, moving across file systems is never atomic.

Reply:
It is possible to specify the directory.